corpdir-srv-jpa/src/main/java/com/github/bordertech/corpdir/jpa/v1/api/ContactServiceImpl.java

```diff
@@ -1,237 +1,32 @@
 package com.github.bordertech.corpdir.jpa.v1.api;
 
-import com.github.bordertech.corpdir.api.exception.NotFoundException;
-import com.github.bordertech.corpdir.api.response.BasicResponse;
-import com.github.bordertech.corpdir.api.response.DataResponse;
-import com.github.bordertech.corpdir.api.v1.ContactService;
-import com.github.bordertech.corpdir.api.v1.model.Channel;
 import com.github.bordertech.corpdir.api.v1.model.Contact;
-import com.github.bordertech.corpdir.api.v1.model.Position;
 import com.github.bordertech.corpdir.jpa.common.map.MapperApiVersion;
 import com.github.bordertech.corpdir.jpa.common.svc.JpaBasicVersionKeyIdService;
-import com.github.bordertech.corpdir.jpa.entity.ChannelEntity;
 import com.github.bordertech.corpdir.jpa.entity.ContactEntity;
-import com.github.bordertech.corpdir.jpa.entity.PositionEntity;
-import com.github.bordertech.corpdir.jpa.entity.VersionCtrlEntity;
 import com.github.bordertech.corpdir.jpa.entity.version.ContactVersionEntity;
-import com.github.bordertech.corpdir.jpa.util.MapperUtil;
 import com.github.bordertech.corpdir.jpa.v1.mapper.ChannelMapper;
 import com.github.bordertech.corpdir.jpa.v1.mapper.ContactMapper;
 import com.github.bordertech.corpdir.jpa.v1.mapper.PositionMapper;
-import java.util.ArrayList;
-import java.util.List;
-import javax.inject.Singleton;
-import javax.persistence.EntityManager;
 
 /**
- * Contact JPA service implementation.
- *
+ * Abstract contact service implementation.
+ * 
  * @author Jonathan Austin
+ * @author Aswin Kandula
  * @since 1.0.0
  */
-@Singleton
-public class ContactServiceImpl extends JpaBasicVersionKeyIdService<Contact, ContactVersionEntity, ContactEntity> implements ContactService {
+public abstract class ContactServiceImpl extends JpaBasicVersionKeyIdService<Contact, ContactVersionEntity, ContactEntity> {
```

Comment:
Could the `service impl` classes implement the `Readonly` and `Write` `API` interfaces to avoid the `abstract` class and the two separate implementations? This way you dont have to split the classes.

If the separate classes are still needed, the packages `jpa.readonly.api.v1` and `jpa.modify.api.v1` should be under `jpa.v1.api`.

Another consideration, to minimise the impact of the split of the `Read` and `Write` interfaces is to create a new `CRUD` interface that implements both `Read` and `Write` and then the `REST` Resources, `JPA` and `DataApis` could use the `CRUD` interface (which is the same as the old service interface).

Your `sync` module can then use the new  `READ` interface for the `FROM` and the `CRUD` as the `TO` as the `TO` needs `READ/WRITE`.

Reply:
@jonathanaustin you are correct I could merge the `Read` and `Write` ServiceImpl (JPA).

For sync module, the merged ServiceImpl (JPA) can become the `TO` and SAP's `READ` impl can become `FROM`, do you think we need a new `CRUD`?

Also for ActionCreator and Store the `API` (CRUD ?) brings the `Read` and `Write` together. Perhaps move this to a `corpdir-srv-api`, so that REST can depend on it too?